docs/USER_GUIDE.md

```diff
@@ -14,14 +33,74 @@ The module names follow the typical hierarchical module
 naming scheme with top-level names like `Control`, `Data`, `System`
 and so on.
 
+
+## Core concepts
+
+### Interfacing linear and non-linear code
+
+Throughout linear code, you will need to interface between non-linear
+and linear code. The key tools you need to do this are in
+[`Data.Unrestricted`] and are typically re-exported by
+[`Prelude.Linear`].
+
+This is basically done through the data definition `Ur` and type classes.
+The data type `Ur` is the identity with an unrestricted/normal arrow on the
+constructor. It basically allows us to ensure that we can use the `a` inside
+an `Ur a` in an unrestricted way.
+
+The type classes `Consumable, Dupable, Moveable` allow non-linear uses of
```

Comment:
To be fair, only `Ur` and `Moveable` are really the interface between linear and non-linear code.

Reply:
Those are for unrestricted code and unrestricted is not equivalent to non-linear. Code that is use at-most-once or at-least-once is still _not_ linear, or, non-linear.

Reply:
Hmm, yes. They are linear ways to do non-linear things. I'll re-name the section and phrasing.